presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraysOverlapFunction.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    @SqlNullable
+    @SqlType(StandardTypes.BOOLEAN)
+    public Boolean arraysOverlapBigInt(
+            @OperatorDependency(operator = LESS_THAN, returnType = StandardTypes.BOOLEAN, argumentTypes = {"bigint", "bigint"}) MethodHandle lessThanFunction,
+            @SqlType("array(bigint)") Block leftArray,
+            @SqlType("array(bigint)") Block rightArray)
+    {
+        int leftSize = leftArray.getPositionCount();
+        int rightSize = rightArray.getPositionCount();
+
+        if (leftSize == 0 || rightSize == 0) {
+            return false;
+        }
+
+        if (arrayBigIntLeft == null || arrayBigIntLeft.length < leftSize) {
+            arrayBigIntLeft = new long[leftSize * 2];
+        }
+        if (arrayBigIntRight == null || arrayBigIntRight.length < rightSize) {
+            arrayBigIntRight = new long[rightSize * 2];
+        }
+
+        boolean leftWithNull = false;
+        int leftRealSize = 0;
+        for (int i = 0; i < leftSize; i++) {
+            if (leftArray.isNull(i)) {
+                leftWithNull = true;
+            }
+            else {
+                arrayBigIntLeft[leftRealSize++] = leftArray.getLong(i, 0);
+            }
+        }
+        sortIfUnsorted(arrayBigIntLeft, leftRealSize);
+
+        boolean rightWithNull = false;
+        int rightRealSize = 0;
+        for (int i = 0; i < rightSize; i++) {
+            if (rightArray.isNull(i)) {
+                rightWithNull = true;
+            }
+            else {
+                arrayBigIntRight[rightRealSize++] = rightArray.getLong(i, 0);
+            }
+        }
+        sortIfUnsorted(arrayBigIntRight, rightRealSize);
+
+        int leftPosition = 0;
+        int rightPosition = 0;
+        while (leftPosition < leftRealSize && rightPosition < rightRealSize) {
+            if (arrayBigIntLeft[leftPosition] < arrayBigIntRight[rightPosition]) {
+                leftPosition++;
+            }
+            else if (arrayBigIntRight[rightPosition] < arrayBigIntLeft[leftPosition]) {
+                rightPosition++;
+            }
+            else {
+                return true;
+            }
+        }
+        return leftWithNull || rightWithNull ? null : false;
+    }
+
+    @SqlNullable
+    @TypeParameter("E")
+    @SqlType(StandardTypes.BOOLEAN)
+    public Boolean arraysOverlap(
+            @OperatorDependency(operator = LESS_THAN, returnType = StandardTypes.BOOLEAN, argumentTypes = {"E", "E"}) MethodHandle lessThanFunction,
+            @TypeParameter("E") Type type,
+            @SqlType("array(E)") Block leftArray,
+            @SqlType("array(E)") Block rightArray)
+    {
+        int leftPositionCount = leftArray.getPositionCount();
+        int rightPositionCount = rightArray.getPositionCount();
+
+        if (leftPositionCount == 0 || rightPositionCount == 0) {
+            return false;
+        }
+
+        if (leftPositions.length < leftPositionCount) {
+            leftPositions = new int[leftPositionCount];
```

Comment:
This can result in degenerative behavior. See my comments for allocating arrays in the other functions.